CODING_STYLE.md

```diff
@@ -0,0 +1,596 @@
+
+       ENV1=... \
+       ENV2=... \
+       require='...' \
+       __type name --state state \
+           --param1 \
+           --param2 foo \
+           --source - <<-'EOF'
+       content sent to ${__object:?}/stdin
+       EOF
+
+   Omitting the parts not needed.
+
+   Multiple parameters may be grouped in a single line if they make sense together, e.g.:
+
+       __file /etc/config-managed \
+           --owner 0 --group 0 --mode 0644 \
+           --source - <<'EOF'
+       This system's configuration is managed by skonfig.
+       EOF
+
+4. **Omit spacing inside input/output redirection operators.**
+
+   It makes it clearer which parts of the command are connected with each other
+   and is consistent with redirection of file descriptors.
+
+       # Don't
+       echo 'hello world' >> /tmp/myfile.txt
+       echo 'hello world' 1 >&2  # prints: hello world 1
+       echo 'hello world' 1> &2  # syntax error
+       echo foo>/tmp/myfile.txt
+       cat >> /tmp/myfile.txt /tmp/a.txt /tmp/b.txt /tmp/c.txt
+
+       # Do
+       echo 'hello world' >>/tmp/myfile.txt
+       echo 'hello world' 1>&2  # prints: hello world
+       echo 'hello world' 1>&2  # prints: hello world
+       echo foo >/tmp/myfile.txt
+       cat >>/tmp/myfile.txt /tmp/a.txt /tmp/b.txt /tmp/c.txt
+
+
+## 5. Comments
+
+1. Comments are introduced by a single `#` character followed by a single space.
+
+2. Comments always obey the 80 character limit.
+   Longer lines are only acceptable if they are example output of a command used
+   to illustrate a point.
+
+3. Comments may use one of these "special" prefixes:
+
+   * "FIXME: "
+   * "NOTE: "
+   * "TODO: "
+
+   If such a prefix is used, all further lines are indented so that the comment
+   lines are aligned.
+
+   Example:
+
+       # just a simple comment
+
+       # NOTE: the following line is special and needs some explanation or there
+       #       is some speciality the reader must know about.
+
+       # FIXME: the following code is broken. This type of comment should only
+       #        be used in very special circumstances. If code is known to be
+       #        broken it should be fixed, not documented.
+
+       # TODO: implement support for operating systems other than Linux.
+
+4. Commented-out code shall not be commited unless there is good reason for it.
+
+
+## 6. Command lists
+
+1. **One command per line only.** Usage of the semicolon (`;`) to separate
+   multiple commands on a single line is strongly discouraged.
+
+   Exception: if having multiple commands on a single line enhances readability,
+   its usage is permissible.
+
+2. If multiple commands are combined using the `&&`, `||`, or `|` (pipe)
+   operators and their length exceeds 80 characters they shall be split to
+   multiple lines with the combination operator as the first on the new line.
+
+   The `&&`, `||`, or `|` operators shall be aligned with the beginning of the
+   previous line.
```

Comment:
What about `if`?

```
if true \
&& false
then
    :
fi
```
or
```
if true \
    && false
then
    :
fi
```

Reply:
Hmm... Good point.
Regarding style I am agnostic. To me both are fine.

I have seen both variants being used in skonfig, with the "indented" version being more common (I think).